### PR TITLE
*: change some call of panic() to error

### PR DESCRIPTION
**Pull request by tiancaiamao**
### What problem does this PR solve? 

Cleanup code

### What is changed and how it works?

change some call of panic() to error

### Check List 

Tests 

 - No code

### CODE REVIEW

**Comment by tiancaiamao** [comment]
PTAL @jackysp @shenli

File: statistics/sample.go [hunk 1]
```diff
@@ -218,7 +217,7 @@ func (s SampleBuilder) CollectColumnStats() ([]*SampleCollector, *SortedBuilder,
 			return collectors, s.PkBuilder, nil
 		}
 		if len(s.RecordSet.Fields()) == 0 {
-			panic(fmt.Sprintf("%T", s.RecordSet))
+			return nil, nil, errors.Errorf("%T", s.RecordSet)
```
Comment (zz-jason) [inline statistics/sample.go]: how about:
```suggestion
			return nil, nil, errors.Errorf("collect column stats failed: record set has 0 field")
```

**Pull request merged**